.evergreen/config.yml

```diff
@@ -1138,6 +1154,20 @@ axes:
           ORCHESTRATION_FILE: "versioned-api-testing.json"
           TEST_FILTER: "VersionedAPITests"
 
+  - id: sanitize
+    display_name: Sanitize
+    values:
+      - id: tsan
+        display_name: TSan
+        variables:
+          SANITIZE: "thread"
+      # ASan is unfortunately known to produce false positives in Swift, see
+      # https://forums.swift.org/t/test-for-memory-leaks-in-ci/36526/9 
```

Comment:
oh yeah, good point. it does work if you copy paste that chunk Johannes wrote into a test somewhere. let me mess around with trying to get that to conditionally run, maybe I can do it based on an environment variable or something like that

Reply:
added new tasks running leaks on macOS. patch here https://spruce.mongodb.com/version/613ba1542a60ed7292b69b91/tasks?sorts=STATUS%3AASC%3BBASE_STATUS%3ADESC

patch where I intentionally added a leak of a `bson_t` to confirm the task would fail: https://spruce.mongodb.com/version/613bae761e2d1720fd5792ef/tasks?sorts=STATUS%3AASC%3BBASE_STATUS%3ADESC

Reply:
ok, fixed up the issues.

I added leak checking tasks on macOS. here's a [successful patch](https://spruce.mongodb.com/version/6140c638a4cf473b2e0b023d/tasks?sorts=STATUS%3AASC%3BBASE_STATUS%3ADESC) and [here's one](https://spruce.mongodb.com/version/6140c66f2a60ed08b9bc0924/tasks?sorts=STATUS%3AASC%3BBASE_STATUS%3ADESC) where memory is intentionally leaked.